packages/opencensus-propagation-stackdriver/test/test-v1.ts

```diff
@@ -0,0 +1,119 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as assert from 'assert';
+import * as mocha from 'mocha';
+import {inspect} from 'util';
+
+import {SpanContext} from '../src/index';
+import {extract, inject} from '../src/v1';
+
+const notNull = <T>(x: T|null|undefined): T => {
+  assert.notStrictEqual(x, null);
+  assert.notStrictEqual(x, undefined);
+  return x as T;
+};
+
+describe('extract', () => {
+  describe('valid inputs', () => {
+    it('should return expected values: cafef00d/667;o=1', () => {
+      const getter = {
+        getHeader() {
+          return 'cafef00d/667;o=1';
+        }
+      };
+      const result = notNull(extract(getter));
+      assert.strictEqual(result.traceId, 'cafef00d');
+      assert.strictEqual(result.spanId, '667');
+      assert.strictEqual(result.options, 1);
+    });
+
+    it('should return expected values:' +
+           '123456/123456123456123456123456123456123456;o=1',
+       () => {
+         const getter = {
+           getHeader() {
+             return '123456/123456123456123456123456123456123456;o=1';
+           }
+         };
+         const result = notNull(extract(getter));
+         assert.strictEqual(result.traceId, '123456');
+         assert.strictEqual(
+             result.spanId, '123456123456123456123456123456123456');
+         assert.strictEqual(result.options, 1);
+       });
+
+    it('should return expected values: 123456/667', () => {
+      const getter = {
+        getHeader() {
+          return '123456/667';
+        }
+      };
+      const result = notNull(extract(getter));
+      assert.strictEqual(result.traceId, '123456');
+      assert.strictEqual(result.spanId, '667');
+      assert.strictEqual(result.options, undefined);
+    });
+  });
+
+  describe('invalid inputs', () => {
+    const inputs = [
+      '', null, undefined, '123456', '123456;o=1', 'o=1;123456', '123;456;o=1',
+      '123/o=1;456', '123/abc/o=1', 'cafefood/667;o=1'
+    ];
+    inputs.forEach(s => {
+      it(`should reject ${s}`, () => {
+        const getter = {
+          getHeader() {
+            // tslint:disable-next-line:no-any
+            return s as any;
+          }
+        };
+        const result = extract(getter);
+        assert.ok(!result);
+      });
+    });
+  });
+});
+
+describe('inject', () => {
+  interface TestData {
+    input: SpanContext;
+    output: string;
+  }
+  const testData: TestData[] = [
+    {
+      input: {traceId: '123456', spanId: '667', options: 1},
+      output: '123456/667;o=1'
+    },
+    {
+      input: {traceId: '123456', spanId: '667', options: undefined},
+      output: '123456/667'
+    },
+    {input: {traceId: '123456', spanId: '667'}, output: '123456/667'}
+  ];
+
+  testData.forEach(({input, output}) => {
+    it(`returns well-formatted trace context for ${inspect(input)}`, () => {
+      const setter = {
+        setHeader(name: string, value: string) {
```

Comment:
Maybe assert that this function will be called?

Reply:
Changed the tests substantially. This is no longer an issue.